### ISSUE TITLE
precompiled release

### ISSUE THREAD

**Issue by mgoeller** [apiel/aboot#1]
Would you mind uploading a precompiled release of aboot.bin?
Thanks

**Comment by mgoeller** [comment]
got it to work, thank you anyways.

**Comment by apiel** [comment]
Hi, may I ask you how come that you want to use this custom bootloader instead of rboot?

**Comment by mgoeller** [comment]
I am trying to make OTA updates work with esp-open-rtos. Using the http-ota example or tftp server. Unfortunately couldn't get it to work reliable yet, saw your issues in the repo and tried it using aboot as a replacement now.

**Comment by apiel** [comment]
Yes, OTA is super unstable. Forget about Tftp since it is using udp. One of the biggest bottleneck is to cleanup space before writing the new firmware `SPIEraseSector`. This is making it crash. So what I was doing is to cleanup the space when the firmware start:

```
extern "C" void user_init(void)
{
    uart_set_baud(0, 115200);

    #ifdef OTA_ENABLE
    ota_prepare();
    #endif
```

```
void ota_prepare(void)
{
    rboot_config rboot_config = rboot_get_config();
    uint8_t nextRom = (rboot_config.current_rom+1)%rboot_config.count;
    printf("ROMROM: current %d count %d offset %d\n", rboot_config.current_rom, rboot_config.count, rboot_config.roms[rboot_config.current_rom]);
    printf("ROMROM: next rom %d next offset %d\n", nextRom, rboot_config.roms[nextRom]);

    #ifndef USE_ABOOT
    printf("ROmROM: cleanup next rom offset.\n");
    if (nextRom != 0) { // we have to keep the first rom, the bootloader will copy the next rom at the first position, so most likely nextRom will always be 1
        int end = rboot_config.roms[nextRom] + 350000; // we assume that our firmware is less than 350 000 bytes
        for(int offset = rboot_config.roms[nextRom]; offset < end; offset += SPI_FLASH_SEC_SIZE) {
            if(sdk_spi_flash_erase_sector(offset/SPI_FLASH_SEC_SIZE) != SPI_FLASH_RESULT_OK) {
                printf("\nsdk_spi_flash_erase_sector stop earlier than plan at %d!\n", offset);
                break;
            }
        }
    }
    #endif
}
```

**Comment by mgoeller** [comment]
I finally managed to get http-ota reliably working with rboot. The main issue was how I flashed the ESP8266. I desoldered an ESP from a NodeMCU and used this to flash many ESP-12E's and F's. That was sometimes successful, sometimes not. Using direct connections from a cheap USB serial adapter worked reliable. Also I had to play around with the --flash-mode settings, if the ESP was not booting I changed it to a lower level, e.g. 'qio' to 'qout', 'dio' or even 'dout'. Thank you anyways. Somehow the flash settings of the of the simple "OTA bootloader firmware" made a difference if the OTA Update via HTTP was working reliable afterwards or not.
